docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1995 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #60.        if max_m &lt; min_m:
</pre><p>Don't do this. Instead, just swap them:</p><pre class="ql-syntax" spellcheck="false">if max_m &lt; min_m:
    min_m, max_m = max_m, min_m
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='64'/>

Reply:
Sure. Just out of curiosity, may I ask why you prefer this rather than letting user know of their mistake?

Reply:
<p>Yes, good question! In Python, there is a subtle concept that is referred to as "it's easier to ask for forgiveness than permission":</p><p> </p><p>https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/</p><p> </p><p>Mind you, in this case, we're not talking about exception handling and, instead, it is more trivial. If we know that the user has made a trivial mistake that we can easily correct for then we should go ahead and do it. The emphasis is on "trivial". Most of the time, the user mistakes are not trivial or easy to interpret and so we can't do anything to help and we shouldn't if we need to make assumptions.</p><p> </p><p>In this particular case, you and I both know that it NEVER makes sense for <code>max_m</code> to be smaller than <code>min_m</code>. In other words, there are ZERO circumstances for <code>max_m &lt; min_m</code> to be <code>True</code> and that any time when that is <code>True</code>, it is 100% clearly a mistake. Now, you could argue that maybe we should at least warn the user that we are swapping the values and I would agree to that because maybe the mistake is <code>min_m=11</code> and <code>max_m=5</code> and the user really meant to do <code>max_m=15</code>. So, a warning would be more appropriate than throwing an error but we still swap the values. </p><p> </p><p>Does that help? In other words, if there is a small chance to "do the right thing" then we should try to do it. This is opportunity is rare though.</p>

Reply:
Thanks for your thorough response and introducing the concept "it's easier to ask for forgiveness than permission" to me.

> I would agree to that because maybe the mistake is `min_m=11` and `max_m=5` and the user really meant to do `max_m=15`. So, a warning would be more appropriate than throwing an error but we still swap the values

Yes...my concern was the typo. So, we can do:
```
#import warnings

if max_m < min_m:
        warnings.warn(f"The provided value for min_m, {min_m}  is more than the provided value for max_m, {max_m}. "
                                f"So, they were swapped.", stacklevel=2)
       
       min_m, max_m = max_m, min_m
```

Since `stacklevel=1` results in the warning mentioning itself at the end of message, I think `stacklevel=2` might be better as it shows the function in which the warning generated. Not sure which is better for the user. 

I can ignore the warning part if that is better for now. If not, please let me know if you have any suggestion about the message.

Reply:
<p>Yes, I think we should ignore the warning part for now. I'm really not concerned about this so much. We perform this swap in other places in the code:</p><p> </p><p>https://github.com/TDAmeritrade/stumpy/blob/b76e852499cb1e97e5e13abc52c4c526fc89687b/stumpy/stimp.py#L175</p><p> </p><p>In fact, it may be better to follow the code above as it eliminates the need for an <code>if</code> check :) </p>